### PR TITLE
fix(supervisor): better error handling and always output final response

### PR DESCRIPTION
**Pull request by suwhang-cisco**
# Description

When context too large, recursion limit etc. is hit, we should always ensure the LLM is aware of the error encountered and still generate the final response to display to the user. In future, we'll also add more safeguarding to reduce the chance of agent hitting this error.

## Type of Change

- [x] Bugfix
- [ ] New Feature
- [ ] Breaking Change
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Pre-release Helm Charts (Optional)

For chart changes, you can test pre-release versions before merging:
- **Base repo contributors:** Create a branch starting with `pre/` for automatic pre-release builds
- **Fork contributors:** Ask a maintainer to add the `helm-prerelease` label
- Pre-release charts are published to `ghcr.io/cnoe-io/pre-release-helm-charts`
- Cleanup happens automatically when the PR closes or label is removed

## Checklist

- [ ] I have read the [contributing guidelines](CONTRIBUTING.md)
- [ ] Existing issues have been referenced (where applicable)
- [ ] I have verified this change is not present in other open pull requests
- [ ] Functionality is documented
- [ ] All code style checks pass
- [ ] New code contribution is covered by automated tests
- [ ] All new and existing tests pass

### CODE REVIEW

File: ai_platform_engineering/multi_agents/platform_engineer/protocol_bindings/a2a/agent.py [hunk 1]
```diff
@@ -272,12 +272,28 @@ async def stream(self, query, context_id, trace_id=None) -> AsyncIterable[dict[s
           logging.error(f"⚠️ Supervisor: Failed to repair orphaned tool calls: {repair_error}")
           # Don't fail - this is a recovery mechanism
 
-      # ========================================================================
-      # SYNTHESIS RETRY CONFIGURATION
-      # If synthesis fails (orphaned tool calls, timeout), retry before failing
-      # ========================================================================
-      max_synthesis_retries = int(os.getenv("MAX_SYNTHESIS_RETRIES", "0"))
-      synthesis_retry_count = 0
+      # ============ TEMPORARY: context overflow test ============
```
Comment (suwhang-cisco) [inline ai_platform_engineering/multi_agents/platform_engineer/protocol_bindings/a2a/agent.py:275]: keeping this as it may come in handy for future edge case testing
